core/src/storage/collections/vec/impls.rs

```diff
@@ -76,12 +78,23 @@ impl<'a, T> Iter<'a, T> {
 
 impl<T> Flush for Vec<T>
 where
-    T: parity_codec::Encode,
+	Self: Encode,
+	T: parity_codec::Encode,
+	T: Flush,
 {
     fn flush(&mut self) {
         self.len.flush();
         self.cells.flush();
     }
+
+    fn flush_at(&mut self, at: Key) {
```

Comment:
Under my impression of how this works we only need an impl of `Flush::flush_at` for `SyncChunk` and all fundamental and primitive types, not for types like `storage::Vec` etc since they already simply propagate their flush to their members.

Reply:
Correct me if I am wrong.